tools/catsitemap.py

```diff
@@ -0,0 +1,119 @@
+"""Script for automatic generation/download of sitemaps for pyansys projects.
+
+Intended for the pyansys project and to be used with the update-gh-pages
+workflow.
+
+"""
+
+from pathlib import Path
+from xml.dom import minidom
+import xml.etree.ElementTree as ET
+
+from links import LINKS
+import requests
+
+
+def download_file(url: str, dest_path: Path) -> None:
+    """Given a sitemap url, this function downloads the file into (dest_path).
+
+    Parameters
+    ----------
+    url : str
+        The url of the sitemap file to be downloaded
+    dest_path : Path
+        The destination path to save the downloaded file
+
+    Raises
+    ------
+    requests.exceptions.Timeout
+        Raises this exception when accessing a link takes too long
+    """
+    # Send the request
+    try:
+        response = requests.get(url, stream=True, timeout=30)
+    except requests.exceptions.Timeout as e:
+        print(f"Timed out while trying to get download the sitemap at this url: {url}")
+        raise e
+
+    # Write the file content to the specified location
+    with open(dest_path, mode="wb") as file:
+        for chunk in response.iter_content(chunk_size=8192):
+            file.write(chunk)
+
+
+def extract_urls_and_headers(links_dict: dict) -> tuple:
+    """Extract valid project names and sitemap urls from metadata dictionary.
+
+    Parameters
+    ----------
+    links_dict : dict
+        Dictionary containing metadata of projects
+
+    Returns
+    -------
+    tuple
+        contains the list of project names and the list of sitemap urls
+    """
+    valid_project_names = []
+    valid_urls = []
+    for project_name, url in links_dict.items():
+        if url is None:
+            continue
+        updated_url = url.split("docs.pyansys.com")[0] + "docs.pyansys.com/sitemap.xml"
```

Comment:
Could you add a comment in here indicating the expected form of `url` and the expected form of `updated_url`?

Reply:
Done!